lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1894,7 +1894,12 @@ def container_create(self, image, create_parameters):
         self.log("create container")
         self.log("image: %s parameters:" % image)
         self.log(create_parameters, pretty_print=True)
-        self.results['actions'].append(dict(created="Created container", create_parameters=create_parameters))
+        # Parameters contains host_config of type docker.types.containers.HostConfig,
+        # derived from dict, which ansible cannot convert to JSON correctly.
+        # Therefore, we make sure we copy it to turn it into a regular dict.
+        debug_parameters = dict(create_parameters)
+        debug_parameters['host_config'] = dict(debug_parameters['host_config'])
```

Comment:
Add a helper for recursive converting anything with an underlying type of `dict`?

Reply:
Is 6c3acc882137f65bce7ada4bfd32cea036182a3f doing what you were thinking of?